### ISSUE TITLE
[mdoc] OutOfMemoryException when running on .NET Framework assemblies

### ISSUE THREAD

**Issue by dend** [mono/api-doc-tools#241]
See binaries in the `binaries` VSTS repo. Error:

```
2018-04-25T20:16:34.1597315Z Updating: System.Web.UI.WebControls.CalendarSelectionMode
2018-04-25T20:16:34.1597315Z Updating: System.Web.UI.WebControls.ICompositeControlDesignerAccessor
2018-04-25T20:16:34.1597315Z Updating: System.Web.UI.WebControls.CompositeControl
2018-04-25T20:16:42.0817706Z Updating: System.Web.UI.WebControls.ChangePassword
2018-04-25T20:16:50.4099093Z ##[error]mdoc: System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at MS.Internal.Xml.XPath.QueryBuilder.ProcessNode(AstNode root, Flags flags, Props& props)
   at MS.Internal.Xml.XPath.QueryBuilder.ProcessOperator(Operator root, Props& props)
   at MS.Internal.Xml.XPath.QueryBuilder.ProcessNode(AstNode root, Flags flags, Props& props)
   at MS.Internal.Xml.XPath.QueryBuilder.ProcessFilter(Filter root, Flags flags, Props& props)
   at MS.Internal.Xml.XPath.QueryBuilder.ProcessNode(AstNode root, Flags flags, Props& props)
   at System.Xml.XPath.XPathExpression.Compile(String xpath, IXmlNamespaceResolver nsResolver)
   at System.Xml.XPath.XPathNavigator.Select(String xpath)
   at System.Xml.XmlNode.SelectNodes(String xpath)
   at Mono.Documentation.MDocUpdater.UpdateSignature(MemberFormatter formatter, MemberReference member, XmlElement xmlElement, FrameworkTypeEntry typeEntry, Boolean fxAlternateTriggered)
   at Mono.Documentation.MDocUpdater.UpdateMember(DocsNodeInfo info, FrameworkTypeEntry typeEntry, Dictionary`2 implementedMembers)
   at Mono.Documentation.MDocUpdater.DoUpdateType2(String message, XmlDocument basefile, TypeDefinition type, FrameworkTypeEntry typeEntry, String output, Boolean insertSince)
   at Mono.Documentation.MDocUpdater.DoUpdateType(TypeDefinition type, FrameworkTypeEntry typeEntry, String basepath, String dest)
   at Mono.Documentation.MDocUpdater.DoUpdateAssembly(AssemblySet assemblySet, AssemblyDefinition assembly, XmlElement index_types, String source, String dest, HashSet`1 goodfiles)
   at Mono.Documentation.MDocUpdater.DoUpdateAssemblies(String source, String dest)
   at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
   at Mono.Documentation.MDoc.Run(String[] args)
   at Mono.Documentation.MDoc.Main(String[] args)
See `mdoc help' for more information or use --debug to diagnose.
```

**Comment by joelmartinez** [comment]
The error also occurred in another location on a different run on a different CI agent

```
System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
at System.Linq.Lookup`2.GetGrouping(TKey key, Boolean create)
at System.Linq.Lookup`2.Create[TSource](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
at System.Linq.GroupedEnumerable`3.GetEnumerator()
at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
at Mono.Documentation.Util.AttachedEntitiesHelper.GetMethodsLookUpTable(TypeDefinition type)
at Mono.Documentation.Util.AttachedEntitiesHelper.<GetAttachedEntities>d__6.MoveNext()
at Mono.Documentation.Util.CecilExtensions.<GetMembers>d__1.MoveNext()
at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
at Mono.Documentation.Updater.DocumentationEnumerator.<GetReflectionMembers>d__5.MoveNext()
at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
at Mono.Documentation.Updater.DocumentationEnumerator.GetMember(TypeDefinition type, DocumentationMember member)
at Mono.Documentation.Updater.DocumentationEnumerator.<GetDocumentationMembers>d__2.MoveNext()
at Mono.Documentation.MDocUpdater.DoUpdateType2(String message, XmlDocument basefile, TypeDefinition type, FrameworkTypeEntry typeEntry, String output, Boolean insertSince)
at Mono.Documentation.MDocUpdater.DoUpdateType(TypeDefinition type, FrameworkTypeEntry typeEntry, String basepath, String dest)
at Mono.Documentation.MDocUpdater.DoUpdateAssembly(AssemblySet assemblySet, AssemblyDefinition assembly, XmlElement index_types, String source, String dest, HashSet`1 goodfiles)
at Mono.Documentation.MDocUpdater.DoUpdateAssemblies(String source, String dest)
at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
at Mono.Documentation.MDoc.Run(String[] args)
at Mono.Documentation.MDoc.Main(String[] args)
```

**Comment by joelmartinez** [comment]
This issue was resolved with the release of mdoc 5.6.5
